### PR TITLE
feat(scheduler): move listers out of algorithm package

### LINKED ISSUE #80372: [Scheduler] Remove dependency of cache on algorithms
`pkg/scheduler/internal/cache` shouldn't depend on `pkg/scheduler/algorithm`

We can move the listers into a a new package `pkg/scheduler/internal/listers` that both `algorithm` and `cache` can depend on.

We can also consider moving `predicates.NodeInfo` into the lister package or another one. This way, `Cache` interface doesn't need to repeat the NodeInfo's functions signatures but rather embed the interface.

/sig scheduling

### PR DESCRIPTION
**Pull request by draveness**
/kind cleanup
/priority important-soon
/assign @ahg-g @alculquicondor 

**What this PR does / why we need it**:

Move listers out of algorithm package

**Which issue(s) this PR fixes**:

Fixes #80372

ref: https://github.com/kubernetes/kubernetes/pull/83882

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

```docs

```

### CODE REVIEW

File: pkg/scheduler/lister/lister.go [hunk 1]
```diff
@@ -1,5 +1,5 @@
 /*
-Copyright 2014 The Kubernetes Authors.
+Copyright 2019 The Kubernetes Authors.
```
Comment (ahg-g) [inline pkg/scheduler/lister/lister.go]: please name the package ```listers``` and the file ```listers.go```

**Review by ahg-g — COMMENTED** [review]
Thanks Draven, one minor nit. 

This is going in to the right direction, but we need to move more listers here in follow up PRs, and make their interface consistent with the one that the k8s client library use so that external entities can create those from the informers cache instead of the scheduler cache if they choose to, I will try to formalize the next steps in a separate issue.

**Comment by draveness** [comment]
I think Golang have a convention of avoiding package name in plural form.

+ https://dmitri.shuralyov.com/idiomatic-go#use-singular-form-for-collection-repo-folder-name
+ https://rakyll.org/style-packages/


Regards,
Draven

**Comment by ahg-g** [comment]
> I think Golang have a convention of avoiding package name in plural form. + https://dmitri.shuralyov.com/idiomatic-go#use-singular-form-for-collection-repo-folder-name + https://rakyll.org/style-packages/ 

but kubernetes goclient name it listers: https://github.com/kubernetes/client-go/blob/master/listers/

**Comment by ahg-g** [comment]
I am trying to cleanup all listers in the scheduler, some of them may not be needed at all, that is why I am delaying this.

**Comment by draveness** [comment]
superseded by https://github.com/kubernetes/kubernetes/pull/84099

/close

**Pull request closed without merging**